examples/reminders/lib/base/data_sources/remote/reminders_firebase_data_source.dart

```diff
@@ -0,0 +1,163 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:rx_bloc_list/models.dart';
+
+import '../../models/reminder/reminder_list_reponse.dart';
+import '../../models/reminder/reminder_model.dart';
+import 'reminders_data_source.dart';
+
+class RemindersFirebaseDataSource implements RemindersDataSource {
+  RemindersFirebaseDataSource() {
+    remindersReference = fireStore.collection('reminders');
+  }
+
+  final fireStore = FirebaseFirestore.instance;
+  late CollectionReference remindersReference;
+
+  ///todo remove it from here
+  int remindersCollectionLength = 0;
+  QueryDocumentSnapshot? lastFetchedRecord;
+
+  late final List<ReminderModel> _data;
+
+  @override
+  Future<ReminderModel> create(
+      {required String title,
+      required DateTime dueDate,
+      required bool complete}) async {
+    var reminder = ReminderModel(
+      dueDate: dueDate,
+      id: remindersReference.doc().id,
+      title: title,
+      complete: complete,
+    );
+
+    var createdReminder = await remindersReference.add(reminder.toJson());
+    var createdReminderId = createdReminder.id;
+    reminder = reminder.copyWith(id: createdReminderId);
+    await remindersReference.doc(createdReminderId).update(reminder.toJson());
```

Comment:
Hmm what is the reason of updating the document right after you have created it ?